### PR TITLE
deprecate k8s 1.14 and 1.15

### PR DESCRIPTION
**Pull request by danielhoherd**
## Description

Deprecate Kube 1.14 and 1.15.

Related to astronomer/issues#2266

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -184,15 +152,15 @@ workflows:
           filters:
             branches:
               only:
-                - '/release-0\.(12|13|14|15|16|23)/'
```
Comment (jpweber) [inline .circleci/config.yml:187]: Was this an accident that the 23 got removed, or is this intentional?

File: .circleci/config.yml [hunk 2]
```diff
@@ -184,15 +152,15 @@ workflows:
...
+                - '/release-0\.(12|13|14|15|16)/'
 
       - release-to-public:
           requires:
             - approve-public-release
           filters:
             branches:
               only:
-                - '/release-0\.(12|13|14|15|16|23)/'
```
Comment (jpweber) [inline .circleci/config.yml:195]: same question here too, intentional change?

**Review by jpweber — CHANGES_REQUESTED** [review]

File: .circleci/config.yml (hunk 1 above)
Comment (danielhoherd) [reply .circleci/config.yml:187]: Oh shoot, yeah, I have no idea how that happened. Good catch!

Comment (danielhoherd) [reply .circleci/config.yml]: The problem was caused by me failing to update the .circleci/config.yml.j2 file when I originally added 23 to the supported list. I committed a change taken from ap-airflow that adds a pre-commit hook to automatically run the generation whenever those files change.

**Review by jpweber — APPROVED** [review]
LGTM

**Review by tommyk — APPROVED** [review]

**Pull request merged**